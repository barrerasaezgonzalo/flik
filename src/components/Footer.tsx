// Footer.tsx

"use client";

import Link from "next/link";
import Image from "next/image";
import React, { useEffect, useState } from "react";
import { FaArrowUp } from "react-icons/fa";

// extraer esta función
export function getScrollY(): number {
  return window.scrollY || document.documentElement.scrollTop;
}

export default function Footer() {
  const [visible, setVisible] = useState(false);

  const toggleVisible = () => {
    const y = getScrollY();
    setVisible(y > 200);
  };

  const scrollToTop = () => {
    window.scrollTo({
      top: 0,
      behavior: "smooth",
    });
  };

  useEffect(() => {
    window.addEventListener("scroll", toggleVisible);
    return () => window.removeEventListener("scroll", toggleVisible);
  }, []);

  return (
    <footer className="bg-gray-100 border-t mt-8 py-6">
      <div className="max-w-4xl mx-auto px-4 flex flex-col gap-6 text-sm text-gray-600">
        <div className="flex flex-col md:flex-row items-center justify-between gap-4">
          <div className="flex items-center gap-3 text-center md:text-left">
            <Link href="/" className="flex items-center gap-2">
              <Image
                src="/logo.png"
                alt="Flik Blog"
                width={48}
                height={48}
                className="w-12 h-12 object-contain"
              />
            </Link>
            <p className="text-gray-600 pt-2">
              Blog de tecnología en español © {new Date().getFullYear()}
            </p>
          </div>

          <nav className="flex flex-wrap items-center gap-4">
            <a href="/about" className="hover:text-gray-800">
              Sobre Flik
            </a>
            <a href="/privacy" className="hover:text-gray-800">
              Privacidad
            </a>
            <a href="/terminos" className="hover:text-gray-800">
              Términos
            </a>
            <a href="/tags" className="hover:text-gray-800">
              Tags
            </a>
            <a href="/contact" className="hover:text-gray-800">
              Contacto
            </a>
            <a
              href="https://www.linkedin.com/company/flikcl/"
              target="_blank"
              className="hover:text-gray-800"
            >
              Linkedin
            </a>
            <a href="/mapa" className="hover:text-gray-800">
              Mapa
            </a>
          </nav>
        </div>
      </div>
      <button
        onClick={scrollToTop}
        aria-label="Volver al inicio"
        className="cursor-pointer fixed bottom-8 right-6 z-50 p-3 bg-green-600 text-white rounded-full shadow-lg hover:bg-green-700 transition"
        style={{ display: visible ? "block" : "none" }}
      >
        <FaArrowUp size={24} />
      </button>
    </footer>
  );
}
