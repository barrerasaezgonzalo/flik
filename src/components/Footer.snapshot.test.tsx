import { render } from "@testing-library/react";
import Footer from "./Footer";
import React from "react";

describe("Footer snapshots", () => {
  it("match snapshot render", () => {
    const { container } = render(<Footer />);
    expect(container).toMatchSnapshot();
  });
});
