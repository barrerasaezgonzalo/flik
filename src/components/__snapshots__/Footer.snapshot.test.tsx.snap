// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer snapshots > match snapshot render 1`] = `
<div>
  <footer
    class="bg-gray-100 border-t mt-8 py-6"
  >
    <div
      class="max-w-4xl mx-auto px-4 flex flex-col md:flex-row items-center justify-between text-sm text-gray-600 gap-4"
    >
      <div
        class="flex items-center gap-3 text-center md:text-left"
      >
        <a
          class="flex items-center gap-2"
          href="/"
        >
          <img
            alt="Flik Blog"
            class="w-12 h-auto"
            height="48"
            src="/logo.png"
            width="48"
          />
          <span
            class="font-semibold text-gray-800"
          >
            Flik
          </span>
        </a>
        <p
          class="text-gray-600"
        >
          Un blog de tecnología aleatoria
        </p>
      </div>
      <nav
        class="flex items-center gap-4"
      >
        <a
          class="hover:text-gray-800"
          href="/about"
        >
          Sobre Flik
        </a>
        <a
          class="hover:text-gray-800"
          href="/privacy"
        >
          Privacidad
        </a>
        <a
          class="hover:text-gray-800"
          href="/terminos"
        >
          Términos
        </a>
        <a
          class="hover:text-gray-800"
          href="/contact"
        >
          Contacto
        </a>
        <a
          class="hover:text-gray-800"
          href="https://www.linkedin.com/company/flikcl/"
          target="_blank"
        >
          Linkedin
        </a>
        <a
          class="hover:text-gray-800"
          href="/mapa"
        >
          Mapa
        </a>
      </nav>
    </div>
    <p
      class="mt-4 text-center text-xs text-gray-700"
    >
      © 
      2025
       Flik
    </p>
  </footer>
</div>
`;
